### PR TITLE
FocusTrapZone - restore last focused descendant element

### PR DESCRIPTION
**Pull request by bworline**
Added FocusTrapZone capability: When FTZ.focus() is called, it will pass focus to a descendant element. The new prop 'focusPreviouslyFocusedInnerElement' controls the descendant-choosing behavior, allowing you to focus either the first descendant or the previously focused descendant.

Unit tests added.

### CODE REVIEW

**Comment by bworline** [comment]
@cliffkoh Can you review?

File: packages/office-ui-fabric-react/src/components/FocusTrapZone/FocusTrapZone.test.tsx [hunk 1]
```diff
@@ -304,4 +304,94 @@ describe('FocusTrapZone', () => {
...
+      setupElement(buttonF, { clientRect: { top: 0, bottom: 10, left: 0, right: 10 } });
+      setupElement(buttonA, { clientRect: { top: 10, bottom: 20, left: 0, right: 10 } });
+      setupElement(buttonB, { clientRect: { top: 20, bottom: 30, left: 0, right: 10 } });
+      setupElement(buttonZ, { clientRect: { top: 30, bottom: 40, left: 0, right: 10 } });
+
+      return { focusTrapZone: focusTrapZoneRef.current!, buttonF, buttonA, buttonB, buttonZ };
+    }
+
+    it('goes to previously focused element when focusing the FTZ', async () => {
+      expect.assertions(4);
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/FocusTrapZone/FocusTrapZone.test.tsx]: :+1:

**Comment by bworline** [comment]
@jspurlin , I removed all auto-focusing when the control gets focus (w.r.t. the previous version of this PR).  Now it only does its thing when .focus() is called.  Can you take a look?

**Comment by bworline** [comment]
@jspurlin is added to the review. #Closed

**Comment by bworline** [comment]
Validation:  Unit tests cover added behavior around focusing when focus() is called.  Checked word-online demo app for regressions in ribbon.

**Pull request merged**